### PR TITLE
[change] ADR-8 document purge deletes

### PR DESCRIPTION
**Pull request by scottf**

### CODE REVIEW

File: adr/ADR-8.md [hunk 1]
```diff
@@ -164,6 +164,9 @@ type KV interface {
 	// Purge removes all data for a key including history, leaving 1 historical entry being the purge
 	Purge(key string) error
 
+	// Completely remove all keys that are in the deleted or purged state, leaving no entries behind for those keys.
+	PurgeDeletes() error
```
Comment (ripienaar) [inline adr/ADR-8.md:168]: We decided not to add this to the required API. So doesn’t really need to be here or should be marked as optional

Comment (scottf) [reply adr/ADR-8.md:168]: Ok I'll make it optional so we can capture the comment as to what it does

**Review by ripienaar — APPROVED** [review]
LGTM

**Review by aricart — COMMENTED** [review]
Just wondering why we don't just overload `Purge(key)` to do this - ie `Purge("")`

**Comment by ripienaar** [comment]
> Just wondering why we don't just overload `Purge(key)` to do this - ie `Purge("")`

This is purge without tombstones that breaks watchers.  Derek was concerned that if someone deletes/purges 100s or 1000s of keys they end up with 1000s of tombstones.  This is essentially a admin maintenance rather than general use function.

That's why its not in the ADR since its largely there to help the CLI offer the facilty.

**Pull request merged**